src/@next/EmptyState/EmptyState.tsx

```diff
@@ -0,0 +1,96 @@
+import React, { useEffect, useState } from 'react';
+import { Button, PrimaryButton } from '../Button';
+import { Neutral } from '../utilities/colors';
+import {
+  EmptyStateContainer,
+  EmptyStateContentContainer,
+  StyledButtonGroup,
+  StyledDescription,
+  StyledHelpText,
+  StyledImage,
+  StyledTitle,
+} from './EmptyStateStyle';
+
+export const imageName = ['empty-carton', 'empty-mailbox', 'safety-cone'];
+export type ImageName = typeof imageName[number];
+export interface EmptyStateProps {
+  buttonBasicContent?: React.ReactNode;
+  buttonBasicLoading?: boolean;
+  buttonBasicOnClick?: React.MouseEventHandler<HTMLButtonElement>;
```

Comment:
Sorry @ninariccimarie I think there's a misunderstanding, we shouldn't pass ReactNode for button content as they could just pass anything which are not related to button. But we should type them as an object e.g: 
{
 label: string;
onClick: (args) => void
}

then on the component we render <Button onClick={() => basicButton.onClick()} >{basicButton.label}</Button>

I think my point to @Afsal102 is if we're passing ReactNode, we don't actually need to worry about the click event as they can just pass it as `<Button onClick={handleClick}>` themselves from the consuming app

Reply:
@marthayaputra i am kind of confused on this statement, from the current implementation the button is already there, we can customise content of the button by passing the props, from ur suggestion if i pass a button again to the `buttonBasicContent`  it will be as follows

<Button> // from the existing
<Button> // pased via the props
</Button>
</Button>

would this be case?

Reply:
may be instead of just getting content of the button, how about we allow the clients to pass the whole button directly ie:

```
 {showBasicButton && (
                {buttonBasic} // we just render whats given from the props
            )}

```

Reply:
@marthayaputra @Afsal102 I'm doing combination of both your initial comments. the reason why I didn't use just label was because of the design requirements.
1. we restrict to only 2 buttons, either primary or basic buttons
2. consuming app can freely pass anything inside those 2 buttons.

point 2 is the reason why i used React.ReactNode.

Reply:
@Afsal102 , what I was saying if the type is  `ReactNode` basically you don't need an even handlers as props, because you can pass construct it in the button you passed, it's like passing a `children`. Jsx is just a sugar syntax for React.functions which are basically bunch of functions with parameters.

Anyway doesn't matter now as @ninariccimarie will change the type to literal objects instead of ReactNode, so now you can define the click function explicitly as prop

Reply:
@ninariccimarie what are the elements they can pass as a button's content? I tot from the last discussion we only limit to `primary` and `secondary`

Reply:
i had it clarified initially. I've tagged both of you on the thread.
I'll apply the changes based on the new requirement which is to only pass label instead of the flexibility that was initially decided.

Reply:
updated these to pass a ButtonAction object @marthayaputra @Afsal102